subsys/bluetooth/audio/ascs.c

```diff
@@ -2033,30 +2033,57 @@ static bool ascs_parse_metadata(struct bt_data *data, void *user_data)
 		return false;
 	}
 
-	/* The CAP acceptor shall not accept metadata with
-	 * unsupported stream context.
-	 */
-	if (IS_ENABLED(CONFIG_BT_CAP_ACCEPTOR)) {
-		if (data_type == BT_AUDIO_METADATA_TYPE_STREAM_CONTEXT) {
-			const uint16_t context = sys_get_le16(data_value);
+	switch (data_type) {
+	/* TODO: Consider rejecting BT_AUDIO_METADATA_TYPE_PREF_CONTEXT type */
+	case BT_AUDIO_METADATA_TYPE_PREF_CONTEXT:
+	case BT_AUDIO_METADATA_TYPE_STREAM_CONTEXT: {
+		uint16_t context;
+
+		if (data_len != sizeof(context)) {
+			*result->rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_METADATA_INVALID,
+						       data_type);
+			result->err = -EBADMSG;
+			return false;
+		}
+
+		context = sys_get_le16(data_value);
+		if (context != (context & BT_AUDIO_CONTEXT_TYPE_ANY)) {
+			*result->rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_METADATA_REJECTED,
+						       data_type);
+			result->err = -EINVAL;
+			return false;
+		}
```

Comment:
Do we really need to verify whether `context & BT_AUDIO_CONTEXT_TYPE_ANY`? IIRC we decided not to assume anything related to assigned numbers in ASCS. 

The only value we care about (if we are the CAP Acceptor) is the `BT_AUDIO_METADATA_TYPE_STREAM_CONTEXT`

Reply:
You know, you convinced me. I check if context is != BT_AUDIO_CONTEXT_TYPE_PROHIBITED only as @MSEEHenrik spotted.
I to not see any ASCS/BAP test cases that would test RFU context type value

Reply:
Test cases BAP/USR/SPE/BI-04-C and BAP/USR/SPE/BI-05-C check for invalid metadata and context values.

Reply:
I saw those in TS but it looks like those are not testing the RFU according to the TS
![image](https://github.com/zephyrproject-rtos/zephyr/assets/10126877/39537e4e-886b-4148-b10a-bfa2405cac4c)
But when I ran the BAP/USR/SPE/BI-04-C it turned out it does. 

Probably it's the matter of wrong test case implementation IMO.
As in 2nd round PTS supposed to write metadata that `Includes incorrectly formatted Metadata value`, but what it does is:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/10126877/fd926643-75f3-4d78-8035-a446e9c0492c)
As you can see the metadata is correctly formatted. I think we all would expect the metadata that has invalid length (which makes if incorrectly formatted) instead of RFU value

Reply:
You are partly right.  When they say incorrectly formatted metadata value, they apparently mean a value that has RFU bits set. Invalid metadata length is covered by test BAP/USR/SPE/BI-02-C.